cg/apps/downsample/downsample.py

```diff
@@ -0,0 +1,166 @@
+"""API that handles downsampling of samples."""
+import logging
+from pathlib import Path
+from typing import List
+
+from cg.constants import SequencingFileTag
+from cg.meta.meta import MetaAPI
+from cg.meta.workflow.downsample.downsample import DownsampleWorkflow
+from cg.models.cg_config import CGConfig
+from cg.models.downsample.downsample_meta_data import DownsampleMetaData
+from cg.store.models import Family, Sample
+from cg.utils.files import get_files_matching_pattern
+
+LOG = logging.getLogger(__name__)
+
+
+class DownSampleAPI(MetaAPI):
+    def __init__(
+        self, config: CGConfig, sample_reads: str, case_internal_id: str, dry_run: bool = False
+    ):
+        """Initialize the API."""
+        super().__init__(config)
+        self.config = config
+        self.sample_reads: str = sample_reads
+        self.case_internal_id: str = case_internal_id
+        self.dry_run: bool = dry_run
+        self.downsample_meta_data: DownsampleMetaData = self.get_meta_data()
```

Comment:
Same here regarding performing the actual function of the class in the instantiation

Reply:
As discussed in the office, if this class cannot be initialised the api should exit.